CONTRIBUTING.md

```diff
@@ -1,196 +1,343 @@
+the tests that run on a pull request, directives can be included in the body:
+
+- To skip all testing (e.g. for a work-in-progress PR), use `[skip ci]`.
+  - This must be included in the title of the pull request, not the body.
+- To run all integration tests, use `[full ci]`.
+- To run _specific_ integration tests, use `[specific ci=$test]`. This will run
+  the specified tests, suites, or groups in addition to the default regression
+  group. Examples:
   - To run the `1-01-Docker-Info` suite: `[specific ci=1-01-Docker-Info]`
   - To run all suites under the `Group1-Docker-Commands` group: `[specific ci=Group1-Docker-Commands]`
   - To run several specific suites: `[specific ci=$test1 --suite $test2 --suite $test3]`.
 - To skip running the unit tests, use `[skip unit]`.
-- To fail fast (make normal failures fatal) during the integration testing, use `[fast fail]`.
-- To specify a specific datastore you want, use `[shared datastore=nfs-datastore]`.
+- To make normal failures fatal during integration testing, use `[fast fail]`.
+- To specify a specific datastore, use `[shared datastore=nfs-datastore]`.
 - To specify the number of parallel jobs you want, use `[parallel jobs=2]`.
 
-You can run the tests locally before making a PR or view the Drone build results for [unit tests and integration tests][dronevic].
+Links to Drone builds results can be found within a pull request and on the list
+of changes pushed to a branch. Results can also be browsed [directly][dronevic].
 
-If you don't have a running ESX required for tests, you can leverage the automated Drone servers for
-running tests. Add `WIP` (work in progress) to the PR title to alert reviewers that the PR is not ready to be merged.
+As the Drone environment is a shared resource, it is best to run tests locally
+before publishing a pull request. If you don't have an environment suitable for
+running the tests, you may leverage the Drone environment. When doing so, it is
+best to include `[WIP]` (work in progress) at the beginning of the title of the
+pull request to alert readers that the change is not ready for review.
 
-If your Drone build needs to be restarted, fork the build:
+Drone builds can be restarted via the web interface or the [CLI][dronecli]:
 ```shell
-export DRONE_TOKEN=<Drone Token>
+export DRONE_TOKEN=<Drone Token, from https://ci-vic.vmware.com/account/token>
 export DRONE_SERVER=https://ci-vic.vmware.com
 
 drone build start vmware/vic <Build Number>
 ```
-If you are not a member of `vmware` org in github, then your PR build may fail. In that case, request one of the existing members / reviewers to fork your failed build to skip membership checking.
+
+For security reasons, your pull request build may fail if you are not a member
+of the `vmware` organization in GitHub. If this occurs, leave a comment on your
+pull request asking that the build be restarted by an organization member.
+
+To skip membership checking when restarting a build:
```

Comment:
when and why would you do this? I assume this is linked to the not above about being a member but it's not completely evident.